### PR TITLE
KAFKA-16231: Update consumer_test.py to support KIP-848’s group protocol config

### PR DESCRIPTION
**Pull request by kirktrue**
Added a new optional `group_protocol` parameter to the test methods, then passed that down to the `setup_consumer` method.

Unfortunately, because the new consumer can only be used with the new coordinator, this required a new `@matrix` block instead of adding the `group_protocol=["classic", "consumer"]` to the existing blocks 😢

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by lucasbru** [comment]
looks good. As you said, we shouldn't merge this yet as long as some are not passing. Could we follow a similar approach as with the integration tests, enable the ones that are passing, and leave the failing ones disabled, ideally linking a ticket?

**Comment by kirktrue** [comment]
@lucasbru can you take another look at this PR? Thanks!

**Comment by kirktrue** [comment]
> Could we follow a similar approach as with the integration tests, enable the ones that are passing, and leave the failing ones disabled, ideally linking a ticket?

Yes, this is the approach I took. Right now, the `all_group_protocols` list (defined in `consumer_group.py`) is—for the time being—just a list of one: the "classic" consumer.

File: tests/kafkatest/tests/client/consumer_test.py [hunk 1]
```diff
@@ -201,7 +200,7 @@ def test_consumer_bounce(self, clean_shutdown, bounce_mode, metadata_quorum=quor
         bounce_mode=["all", "rolling"],
         num_bounces=[5],
         metadata_quorum=[quorum.isolated_kraft],
-        use_new_coordinator=[True, False]
+        use_new_coordinator=[True]
```
Comment (lucasbru) [inline tests/kafkatest/tests/client/consumer_test.py]: Why no group protocol parameter here?

File: tests/kafkatest/tests/verifiable_consumer_test.py [hunk 2]
```diff
@@ -72,16 +86,59 @@ def await_produced_messages(self, producer, min_messages=1000, timeout_sec=10):
...
     def await_members(self, consumer, num_consumers):
         # Wait until all members have joined the group
-        wait_until(lambda: len(consumer.joined_nodes()) == num_consumers,
-                   timeout_sec=self.session_timeout_sec*2,
-                   err_msg="Consumers failed to join in a reasonable amount of time")
-        
+        states = [ConsumerState.Joined]
+
+        if consumer_group.is_consumer_group_protocol_enabled(consumer.group_protocol):
+            states.extend([ConsumerState.Started, ConsumerState.Rebalancing])
```
Comment (lucasbru) [inline tests/kafkatest/tests/verifiable_consumer_test.py]: Why do we have to do this? Are we still checking the same thing if we consider "started" as "joined"?

**Review by lucasbru — COMMENTED** [review]
Hey @kirktrue, thanks for PR!
 - You changed some of the testing code. Are these just refactorings? Please leave some comments why we are changing the code around waiting for states and assignments. I don't want to stop you from cleaning up things, I just have a hard time following the changes.
 - Why do we only add the group protocol parameter in some of the tests?
 - Why do you want to commit this without adding the `consumer` group in the set of valid group protocols. Are none of the tests passing right now? Feels like we are adding dead code.

**Review by lucasbru — APPROVED** [review]

**Comment by lucasbru** [comment]
LGTM, thanks!

**Pull request merged**